hazelcast/src/main/java/com/hazelcast/internal/affinity/ThreadAffinity.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.affinity;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target(TYPE)
+public @interface ThreadAffinity {
```

Comment:
I have no hard preference for an annotation or for an overridable method on the HZManagedThread called 'setPoolName' which can be called in the constructor of PartitionThread and IOThread. I think this is simpler than introducing an annotation to carry the same information; and which can be used incorrectly (placing it on a non HZManagedThread for example and there won't be any feedback).